apps/blinkt/device.c

```diff
@@ -1198,7 +1198,7 @@ int Device_Read_Property_Local(BACNET_READ_PROPERTY_DATA *rpdata)
  * @return The length of the APDU on success, else BACNET_STATUS_ERROR
  */
 static int Read_Property_Common(
-    struct object_functions *pObject, BACNET_READ_PROPERTY_DATA *rpdata)
+    const struct object_functions *pObject, BACNET_READ_PROPERTY_DATA *rpdata)
```

Comment:
rpdata could also be const'ed.

Reply:
Cannot yet be as we write to rpdata->application_data inside property_list_common_encode().

I highly recommend using Sonarlint for catching these. You just need to add

`set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`

to cmake and configure path to compile_commands.json

Also note that this PR makes like 80% const correct. I do not want to make all things here as things get more tricky as we need to seperete some functions to read and write functions to get all things done. I only do easy steps here.